src/bootstrap/config.rs

```diff
@@ -782,14 +771,75 @@ impl Config {
             set(&mut config.missing_tools, t.missing_tools);
         }
 
+        // This is duplicated in bootstrap.py.
+        config.out = build.build_dir.unwrap_or_else(|| PathBuf::from("build"));
+        t!(fs::create_dir_all(&config.out));
+        config.out = config.out.canonicalize().expect("Build directory canonicalized");
+        config.out = config
+            .out
+            .components()
+            .flat_map(|component| {
+                if component.as_os_str() == "$ROOT" {
```

Comment:
This... seems like a misfeature. I guess it's supported in the python code though...

Reply:
This was added in https://github.com/rust-lang/rust/pull/71994, but yeah I'm not entirely 100% sure it's something we truly need (vs. forcing absolute paths or whatever in configuration).

Reply:
I'd be ok with removing this. What I'd _really_ like is for the build directory to always be relative to the source directory, not the current directory, but I don't think anyone is using `$ROOT`.